addons/web/static/tests/model/sample_server.test.js

```diff
@@ -0,0 +1,444 @@
+        arrival_date: { string: "Date of arrival", type: "datetime" },
+        height: { string: "Height", type: "float" },
+        color: { string: "Color", type: "integer" },
+        age: { string: "Age", type: "integer" },
+        salary: { string: "Salary", type: "monetary" },
+        currency: {
+            string: "Currency",
+            type: "many2one",
+            relation: "res.currency",
+        },
+        manager_id: {
+            string: "Manager",
+            type: "many2one",
+            relation: "res.users",
+        },
+        cover_image_id: {
+            string: "Cover Image",
+            type: "many2one",
+            relation: "ir.attachment",
+        },
+        managed_ids: {
+            string: "Managing",
+            type: "one2many",
+            relation: "res.users",
+        },
+        tag_ids: { string: "Tags", type: "many2many", relation: "tag" },
+        type: {
+            string: "Type",
+            type: "selection",
+            selection: [
+                ["client", "Client"],
+                ["partner", "Partner"],
+                ["employee", "Employee"],
+            ],
+        },
+    },
+    "res.country": {
+        display_name: { string: "Name", type: "char" },
+    },
+    hobbit: {
+        display_name: { string: "Name", type: "char" },
+        profession: {
+            string: "Profession",
+            type: "selection",
+            selection: [
+                ["gardener", "Gardener"],
+                ["brewer", "Brewer"],
+                ["adventurer", "Adventurer"],
+            ],
+        },
+        age: { string: "Age", type: "integer" },
+    },
+    "ir.attachment": {
+        display_name: { string: "Name", type: "char" },
+    },
+};
+
+describe("Sample data", () => {
+    test("people type + all field names", async () => {
+        const specification = {};
+        for (const fieldName in fields["res.users"]) {
+            specification[fieldName] = {};
+            if (fields["res.users"][fieldName].type === "many2one") {
+                specification[fieldName] = {
+                    fields: { display_name: {} },
+                };
+            }
+        }
+        const server = new DeterministicSampleServer("res.users", fields["res.users"]);
+        const { records } = await server.mockRpc({
+            method: "web_search_read",
+            model: "res.users",
+            specification,
+        });
+        const rec = records[0];
+        function expectFormat(fieldName, regex) {
+            if (regex instanceof RegExp) {
+                expect(regex.test(rec[fieldName].toString())).toBe(true, {
+                    message: `Field "${fieldName}" has the correct format`,
+                });
+            } else {
+                expect(typeof rec[fieldName]).toBe(regex, {
+                    message: `Field "${fieldName}" is of type ${regex}`,
+                });
+            }
+        }
```

Comment:
maybe use the right expect where and when it is needed?
`expect(rec.fieldName.toString()).toMatch(regex)`
`expect(rec.fieldName).toBeOfType(type)`